custom_components/tahoma/select.py

```diff
@@ -33,40 +42,48 @@ async def async_setup_entry(
     data = hass.data[DOMAIN][entry.entry_id]
     coordinator = data["coordinator"]
 
-    entities = [
-        PedestrianGateSelect(device.device_url, coordinator)
-        for device in data["platforms"][COVER]
-        if CORE_OPEN_CLOSED_PEDESTRIAN_STATE in device.states
-    ]
+    entities = []
 
-    async_add_entities(entities)
+    key_supported_states = {
+        description.key: description for description in SELECT_DESCRIPTIONS
+    }
 
+    for device in coordinator.data.values():
+        for state in device.definition.states:
+            if description := key_supported_states.get(state.qualified_name):
+                print("added")
+                print(state.qualified_name)
+                entities.append(
+                    OverkizSelect(
+                        device.device_url,
+                        coordinator,
+                        description,
+                    )
+                )
 
-class PedestrianGateSelect(OverkizEntity, SelectEntity):
-    """Representation of the various state for a pedestrian gate."""
+    async_add_entities(entities)
 
-    _attr_icon = "mdi:content-save-cog"
 
-    def __init__(
-        self,
-        device_url: str,
-        coordinator: OverkizDataUpdateCoordinator,
-    ):
-        """Initialize the device."""
-        super().__init__(device_url, coordinator)
-        self._attr_name = f"{super().name} Position"
+class OverkizSelect(OverkizDescriptiveEntity, SelectEntity):
+    """Representation of an Overkiz Number entity."""
 
     @property
     def current_option(self):
         """Return the selected entity option to represent the entity state."""
-        return self.device.states.get(CORE_OPEN_CLOSED_PEDESTRIAN_STATE).value
+        state = self.device.states.get(self.entity_description.key)
```

Comment:
Would be nice to use same code that the one in Number

Reply:
Will have a look! I took this from sensor.oh

Reply:
Fixed.